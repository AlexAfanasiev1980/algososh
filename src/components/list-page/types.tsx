export interface IData {
  number?: number | undefined | null;
  index?: number | undefined;
  type: "head" | "tail" | "delHead" | "delTail" | "AddIndex" | "DelIndex" | "";
}