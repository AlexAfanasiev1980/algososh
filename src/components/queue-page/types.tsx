import { ElementStates } from "../../types/element-states";

export interface IColor {
  color?: ElementStates;
  index?: number | undefined;
}

export interface IButtonState {
  add: boolean
  delete: boolean
  clear: boolean
}