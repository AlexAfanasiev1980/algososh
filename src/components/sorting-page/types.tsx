export interface IElements {
  arr: Array<number[]>;
  comparable: Array<number[]>;
  notsorted: Array<number[]>;
  sorted: Array<number[]>;
}

export interface IElement {
  arr: number[];
  comparable: number[];
  notsorted: number[];
  sorted: number[];
}