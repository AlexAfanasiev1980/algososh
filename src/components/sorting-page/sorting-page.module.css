.flex {
  display: flex;
  justify-content: flex-start;
}

.itemList {
  max-width: 980px;
  margin: 0 auto;
}

.radioInput {
  width: 221px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 0 52px 0 0;
}

.button {
  width: 422px;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.button_rigth {
  margin: 0 0 0 80px;
}

.column {
  height: 347px;
  align-items: flex-end;
  justify-content: center;
  max-width: 980px;
  margin: 48px auto 0 auto;
}