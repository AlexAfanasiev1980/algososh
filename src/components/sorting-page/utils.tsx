export const swap = (
  arr: number[],
  firstIndex: number,
  secondIndex: number
): number[] => {
  const temp = arr[firstIndex];
  arr[firstIndex] = arr[secondIndex];
  arr[secondIndex] = temp;
  return arr;
};