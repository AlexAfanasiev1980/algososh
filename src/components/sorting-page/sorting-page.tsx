import React, { useState, useEffect } from "react";
import { SolutionLayout } from "../ui/solution-layout/solution-layout";
import { RadioInput } from "../ui/radio-input/radio-input";
import { Button } from "../ui/button/button";
import { Column } from "../ui/column/column";
import styles from "./sorting-page.module.css";
import { Direction } from "../../types/direction";
import { ElementStates } from "../../types/element-states";
import { IElements, IElement, IButtonState } from "./types";
import { SHORT_DELAY_IN_MS } from "../../constants/delays";
import { swap, defaultState, defaultButtonState } from "./utils";

export const handleSelectionSort = (choice: number, arrayNew: number[], array: number[], elements: IElements) => {
  const { length } = array;
  const arrayElemets: Array<number[]> = [arrayNew];
  let elementsData: IElements = elements;
  let sortedData: number[] = [];
  let comparableData: number[] = [];
  let notsortedData: number[] = [];
  let step: number = 0;
  for (let i = 0; i < length - 1; i++) {
    let maxInd = i;
    let index = i;
    let minInd: number | undefined = 0;
    for (let j = maxInd + 1; j <= length - 1; j++) {
      for (let u = 0; u < length; u++) {
        const filter: number[] = elementsData.sorted[step].filter(
          (el) => u === el
        );
        if (u !== i && u !== j && filter.length === 0) {
          notsortedData.push(u);
        }
      }
      elementsData = {
        ...elementsData,
        arr: [
          ...elementsData.arr,
          elementsData.arr[elementsData.arr.length - 1],
        ],
        comparable: [...elementsData.comparable, [i, j]],
        notsorted: [...elementsData.notsorted, [...notsortedData]],
        sorted: [...elementsData.sorted, [...elementsData.sorted[step]]],
      };
      notsortedData = [];
      step++;
      if (choice === 1) {
        if (array[j] < array[index]) {
          minInd = j;
          index = j;
        }
      } else {
        if (array[j] > array[index]) {
          minInd = j;
          index = j;
        }
      }
    }
    let arrAdd: number[];
    sortedData.push(i);
    if (minInd !== 0) {
      arrAdd = swap(array, maxInd, minInd);
    } else {
      arrAdd = elementsData.arr[step];
    }
    arrayElemets.push([...arrAdd]);
    if (length - 2 <= i) {
      sortedData.push(i + 1);
    } else {
      comparableData = [i + 1, i + 2];
    }
    for (let u = 0; u < length; u++) {
      const filterSort = sortedData.includes(u);
      const filterComparable = comparableData.includes(u);
      if (u !== i && !filterSort && !filterComparable) {
        notsortedData.push(u);
      }
    }
    elementsData = {
      ...elementsData,
      arr: [...elementsData.arr, [...arrAdd]],
      comparable: [...elementsData.comparable, [...comparableData]],
      notsorted: [...elementsData.notsorted, [...notsortedData]],
      sorted: [...elementsData.sorted, [...sortedData]],
    };
    step++;
    notsortedData = [];
  }

  return elementsData;
}

export const handleBubbleSort = (choice: number, arrayNew: number[], array: number[], elements: IElements) => {
  const { length } = array;
  let arrayElemets: number[] = arrayNew;
    let elementsData: IElements = elements;
    let sortedData: number[] = [];
    let notsortedData: number[] = [];
    let step: number = 0;

    for (let i = 0; i < length - 1; i++) {
      for (let j = 0; j < length - 1 - i; j++) {
        for (let u = 0; u < length; u++) {
          const filter: number[] = elementsData.sorted[step].filter(
            (el) => u === el
          );
          if (u !== j + 1 && u !== j && filter.length === 0) {
            notsortedData.push(u);
          }
        }
        elementsData = {
          ...elementsData,
          arr: [...elementsData.arr, [...arrayElemets]],
          comparable: [...elementsData.comparable, [j, j + 1]],
          notsorted: [...elementsData.notsorted, [...notsortedData]],
          sorted: [...elementsData.sorted, [...sortedData]],
        };
        notsortedData = [];
        step++;
        if (choice === 1) {
          if (arrayElemets[j + 1] < arrayElemets[j]) {
            arrayElemets = [...swap(arrayElemets, j, j + 1)];
            elementsData = {
              ...elementsData,
              arr: [...elementsData.arr, [...arrayElemets]],
              comparable: [
                ...elementsData.comparable,
                [...elementsData.comparable[step]],
              ],
              notsorted: [
                ...elementsData.notsorted,
                [...elementsData.notsorted[step]],
              ],
              sorted: [...elementsData.sorted, [...elementsData.sorted[step]]],
            };
            step++;
          }
          if (j === length - 2 - i) {
            sortedData.push(j + 1);
            for (let u = 0; u < length - 1 - i; u++) {
              if (u !== j) {
                notsortedData.push(u);
              }
            }
            elementsData = {
              ...elementsData,
              arr: [...elementsData.arr, [...arrayElemets]],
              comparable: [...elementsData.comparable, []],
              notsorted: [...elementsData.notsorted, [...notsortedData]],
              sorted: [...elementsData.sorted, [...sortedData]],
            };
            notsortedData = [];
            step++;
          }
          if (i === length - 2) {
            sortedData.push(0);
            elementsData = {
              ...elementsData,
              arr: [...elementsData.arr, [...arrayElemets]],
              comparable: [...elementsData.comparable, []],
              notsorted: [...elementsData.notsorted, []],
              sorted: [...elementsData.sorted, [...sortedData]],
            };
          }
        } else {
          if (arrayElemets[j + 1] > arrayElemets[j]) {
            arrayElemets = [...swap(arrayElemets, j, j + 1)];
            elementsData = {
              ...elementsData,
              arr: [...elementsData.arr, [...arrayElemets]],
              comparable: [
                ...elementsData.comparable,
                [...elementsData.comparable[step]],
              ],
              notsorted: [
                ...elementsData.notsorted,
                [...elementsData.notsorted[step]],
              ],
              sorted: [...elementsData.sorted, [...elementsData.sorted[step]]],
            };
            step++;
          }
          if (j === length - 2 - i) {
            sortedData.push(j + 1);
            for (let u = 0; u < length - 1 - i; u++) {
              if (u !== j) {
                notsortedData.push(u);
              }
            }
            elementsData = {
              ...elementsData,
              arr: [...elementsData.arr, [...arrayElemets]],
              comparable: [...elementsData.comparable, []],
              notsorted: [...elementsData.notsorted, [...notsortedData]],
              sorted: [...elementsData.sorted, [...sortedData]],
            };
            notsortedData = [];
            step++;
          }
          if (i === length - 2) {
            sortedData.push(0);
            elementsData = {
              ...elementsData,
              arr: [...elementsData.arr, [...arrayElemets]],
              comparable: [...elementsData.comparable, []],
              notsorted: [...elementsData.notsorted, []],
              sorted: [...elementsData.sorted, [...sortedData]],
            };
          }
        }
      }
    }
    return elementsData;
}


export const SortingPage: React.FC = () => {
  const [buttonState, setButtonState] =
    useState<IButtonState>(defaultButtonState);
  const [getCheckedsetChoice, setCheckedChoice] = useState<boolean>(true);
  const [getCheckedVial, setCheckedVial] = useState<boolean>(false);
  const [array, setArray] = useState<number[]>([]);
  const [elements, setElements] = useState<IElements>(defaultState);
  const [element, setElement] = useState<IElement>(defaultState);
  const [arrayNew, setArrayNew] = useState<number[]>([]);

  const randomArr = (e: React.SyntheticEvent): void => {
    e.preventDefault();
    const random = Math.floor(Math.random() * (18 - 3) + 3);
    const arr = [];
    for (let i = 0; i < random; i++) {
      arr.push(Math.floor(Math.random() * 100));
    }
    const notsorted = [];
    for (let i = 0; i < arr.length; i++) {
      notsorted.push(i);
    }
    setArray([...arr]);
    setArrayNew([...arr]);
    setElements({
      ...elements,
      arr: [arr],
      notsorted: [notsorted],
      comparable: [[]],
      sorted: [[]],
    });
  };

  const selectionSort = (choice: number) => {
    const notsorted = [];
    for (let i = 0; i < array.length; i++) {
      notsorted.push(i);
    }
    setElements({
      ...elements,
      arr: [array],
      notsorted: [notsorted],
      comparable: [[]],
      sorted: [[]],
    });
    setElements(handleSelectionSort(choice, arrayNew, array, elements))
  };

  const BubbleSort = (choice: number) => {
    setElements({
      ...elements,
      arr: [],
      notsorted: [],
      comparable: [],
      sorted: [],
    });
    const notsorted = [];
    for (let i = 0; i < array.length; i++) {
      notsorted.push(i);
    }
    setElements({
      ...elements,
      arr: [array],
      notsorted: [[...notsorted]],
      comparable: [[]],
      sorted: [[]],
    });
    setElements(handleBubbleSort(choice, arrayNew, array, elements));
  };

  const handleButton = (e: React.SyntheticEvent): void => {
    e.preventDefault();
    if (element.arr.length === 0) {
      return
    }
    const input = e.target as HTMLElement;
    setButtonState((prevState) => {
      return {
        ...prevState,
        upButton: {
          disabled: input.innerText === "По убыванию",
          loading: input.innerText === "По возрастанию",
        },
        downButton: {
          disabled: input.innerText === "По возрастанию",
          loading: input.innerText !== "По возрастанию",
        },
        arrButton: {
          disabled: true,
        },
      };
    });

    getCheckedsetChoice
      ? input.innerText === "По возрастанию"
        ? selectionSort(1)
        : selectionSort(2)
      : input.innerText === "По возрастанию"
      ? BubbleSort(1)
      : BubbleSort(2);
  };

  const handleInput = (e: React.SyntheticEvent): void => {
    const input = e.target as HTMLInputElement;
    if (input.value === "choice") {
      setCheckedChoice(true);
      setCheckedVial(false);
    } else {
      setCheckedChoice(false);
      setCheckedVial(true);
    }
  };

  const count1 = (steps: number, count: number) => {
    setElement({
      ...element,
      arr: elements.arr[count],
      notsorted: elements.notsorted[count],
      comparable: elements.comparable[count],
      sorted: elements.sorted[count],
    });
    count++;
    if (count > steps - 1) {
      setButtonState(defaultButtonState);
      return;
    } else {
      setTimeout(count1, SHORT_DELAY_IN_MS, steps, count);
    }
  };

  useEffect(() => {
    if (elements.arr.length > 0) {
      setTimeout(count1, SHORT_DELAY_IN_MS, elements.arr.length, 0);
    }
  }, [elements]);

  return (
    <SolutionLayout title="Сортировка массива">
      <div>
        <div className={`${styles.flex} ${styles.itemList}`}>
          <div className={`${styles.flex} ${styles.radioInput}`}>
            <RadioInput
              label="Выбор"
              name="choice"
              value="choice"
              checked={getCheckedsetChoice}
              onChange={handleInput}
            />
            <RadioInput
              label="Пузырек"
              name="vial"
              value="vial"
              checked={getCheckedVial}
              onChange={handleInput}
            />
          </div>
          <div className={`${styles.flex} ${styles.button}`}>
            <Button
              text="По возрастанию"
              sorting={Direction.Ascending}
              onClick={handleButton}
              isLoader={buttonState.upButton.loading}
              linkedList="average"
              disabled={buttonState.upButton.disabled}
            />
            <Button
              text="По убыванию"
              sorting={Direction.Descending}
              onClick={handleButton}
              isLoader={buttonState.downButton.loading}
              linkedList="average"
              disabled={buttonState.downButton.disabled}
            />
          </div>
          <div className={`${styles.button_rigth}`}>
            <Button
              text="Новый массив"
              linkedList="average"
              onClick={randomArr}
              disabled={buttonState.arrButton.disabled}
            />
          </div>
        </div>
        <div className={`${styles.flex} ${styles.column}`}>
          {element.arr &&
            element.arr.map((el, index) => {
              let color: ElementStates = ElementStates.Default;
              const filterSorted = element.sorted.includes(index);
              const filterNotSorted = element.notsorted.includes(index);
              const filterComparable = element.comparable.includes(index);
              if (filterSorted) {
                color = ElementStates.Modified;
              } else if (filterNotSorted) {
                color = ElementStates.Default;
              } else if (filterComparable) {
                color = ElementStates.Changing;
              }
              return (
                <Column
                  index={el}
                  extraClass="ml-2"
                  state={color}
                  key={`${el.toString()}${index.toString()}`}
                />
              );
            })}
        </div>
      </div>
    </SolutionLayout>
  );
};
