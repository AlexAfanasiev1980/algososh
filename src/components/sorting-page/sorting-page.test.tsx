import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { SortingPage } from "./sorting-page";
import { cleanup } from "@testing-library/react";
import { handleSelectionSort, handleBubbleSort } from "./sorting-page";
import { IElements, IElement, IButtonState } from "./types";


it("should render and Match Snapshot", () => {
  const ProductComponent = renderer.create(
    <Router>
      <SortingPage />
    </Router>
  );
  expect(ProductComponent).toMatchSnapshot();
});

afterEach(cleanup);

describe("Тестирование сортировки выбором", () => {
  it("Сортировка массива выбором по возрастанию работает корректно несколько элементов", () => {
    const sortedArr: IElements = handleSelectionSort(1, [25, 5, 10, 45], [25, 5, 10, 45], {
      arr: [[25, 5, 10, 45]],
      notsorted: [[0,1,2,3]],
      comparable: [[]],
      sorted: [[]]
    });
    const arr = sortedArr.arr;
    expect(arr[arr.length-1]).toStrictEqual([5, 10, 25, 45]);
  });
  it("Сортировка массива выбором по убыванию работает корректно несколько элементов", () => {
    const sortedArr: IElements = handleSelectionSort(2, [25, 5, 10, 45], [25, 5, 10, 45], {
      arr: [[25, 5, 10, 45]],
      notsorted: [[0,1,2,3]],
      comparable: [[]],
      sorted: [[]]
    });
    const arr = sortedArr.arr;
    expect(arr[arr.length-1]).toStrictEqual([45, 25, 10, 5]);
  });
  it("Сортировка массива выбором по возрастанию работает корректно один элементов", () => {
    const sortedArr: IElements = handleSelectionSort(1, [25], [25], {
      arr: [[25]],
      notsorted: [[0]],
      comparable: [[]],
      sorted: [[]]
    });
    const arr = sortedArr.arr;
    expect(arr[arr.length-1]).toStrictEqual([25]);
  });
  it("Сортировка массива выбором по убыванию работает корректно один элементов", () => {
    const sortedArr: IElements = handleSelectionSort(2, [25], [25], {
      arr: [[25]],
      notsorted: [[0]],
      comparable: [[]],
      sorted: [[]]
    });
    const arr = sortedArr.arr;
    expect(arr[arr.length-1]).toStrictEqual([25]);
  });
  it("Сортировка массива выбором по возрастанию работает корректно пустой массив", () => {
    const sortedArr: IElements = handleSelectionSort(1, [], [], {
      arr: [[]],
      notsorted: [[]],
      comparable: [[]],
      sorted: [[]]
    });
    const arr = sortedArr.arr;
    expect(arr[arr.length-1]).toStrictEqual([]);
  });
  it("Сортировка массива выбором по убыванию работает корректно пустой массив", () => {
    const sortedArr: IElements = handleSelectionSort(2, [], [], {
      arr: [[]],
      notsorted: [[]],
      comparable: [[]],
      sorted: [[]]
    });
    const arr = sortedArr.arr;
    expect(arr[arr.length-1]).toStrictEqual([]);
  });
});

describe("Тестирование сортировки пузырьком", () => {
  it("Сортировка массива пузырьком по возрастанию работает корректно несколько элементов", () => {
    const sortedArr: IElements = handleBubbleSort(1, [25, 5, 10, 45], [25, 5, 10, 45], {
      arr: [[25, 5, 10, 45]],
      notsorted: [[0,1,2,3]],
      comparable: [[]],
      sorted: [[]]
    });
    const arr = sortedArr.arr;
    expect(arr[arr.length-1]).toStrictEqual([5, 10, 25, 45]);
  });
  it("Сортировка массива пузырьком по убыванию работает корректно несколько элементов", () => {
    const sortedArr: IElements = handleBubbleSort(2, [25, 5, 10, 45], [25, 5, 10, 45], {
      arr: [[25, 5, 10, 45]],
      notsorted: [[0,1,2,3]],
      comparable: [[]],
      sorted: [[]]
    });
    const arr = sortedArr.arr;
    expect(arr[arr.length-1]).toStrictEqual([45, 25, 10, 5]);
  });
  it("Сортировка массива пузырьком по возрастанию работает корректно один элементов", () => {
    const sortedArr: IElements = handleBubbleSort(1, [25], [25], {
      arr: [[25]],
      notsorted: [[0]],
      comparable: [[]],
      sorted: [[]]
    });
    const arr = sortedArr.arr;
    expect(arr[arr.length-1]).toStrictEqual([25]);
  });
  it("Сортировка массива пузырьком по убыванию работает корректно один элементов", () => {
    const sortedArr: IElements = handleBubbleSort(2, [25], [25], {
      arr: [[25]],
      notsorted: [[0]],
      comparable: [[]],
      sorted: [[]]
    });
    const arr = sortedArr.arr;
    expect(arr[arr.length-1]).toStrictEqual([25]);
  });
  it("Сортировка массива пузырьком по возрастанию работает корректно пустой массив", () => {
    const sortedArr: IElements = handleBubbleSort(1, [], [], {
      arr: [[]],
      notsorted: [[]],
      comparable: [[]],
      sorted: [[]]
    });
    const arr = sortedArr.arr;
    expect(arr[arr.length-1]).toStrictEqual([]);
  });
  it("Сортировка массива пузырьком по убыванию работает корректно пустой массив", () => {
    const sortedArr: IElements = handleBubbleSort(2, [], [], {
      arr: [[]],
      notsorted: [[]],
      comparable: [[]],
      sorted: [[]]
    });
    const arr = sortedArr.arr;
    expect(arr[arr.length-1]).toStrictEqual([]);
  });
});
