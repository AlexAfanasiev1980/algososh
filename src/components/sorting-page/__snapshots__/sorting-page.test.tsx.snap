// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render and Match Snapshot 1`] = `
<main
  className="content "
>
  <div
    className="titleBox"
  >
    <h1
      className="text text_type_h2 text_color_h1 title"
    >
      МБОУ АЛГОСОШ
    </h1>
    <span
      className="text text_type_fibonacci text_color_secondary subtitle"
    >
      им. Фибоначчи
    </span>
  </div>
  <div
    className="contentCard"
  >
    <a
      className="link"
      href="/"
      onClick={[Function]}
    >
      <button
        className="button returnButton"
        type="button"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M12.7073 4.29289C13.0978 4.68342 13.0978 5.31658 12.7073 5.70711L7.41436 11H19.0002C19.5524 11 20.0002 11.4477 20.0002 12C20.0002 12.5523 19.5524 13 19.0002 13H7.41436L12.7073 18.2929C13.0978 18.6834 13.0978 19.3166 12.7073 19.7071C12.3167 20.0976 11.6836 20.0976 11.293 19.7071L3.58594 12L11.293 4.29289C11.6836 3.90237 12.3167 3.90237 12.7073 4.29289Z"
            fill="#0032FF"
            fillRule="evenodd"
          />
        </svg>
        <p
          className="text text_type_button text_color_link ml-4"
        >
          К оглавлению
        </p>
      </button>
    </a>
    <h3
      className="text text_type_h3 text_color_h3 cardTitle"
    >
      Сортировка массива
    </h3>
    <div>
      <div
        className="flex itemList"
      >
        <div
          className="flex radioInput"
        >
          <div
            className="content "
          >
            <input
              checked={true}
              className="input"
              id="igRc8KcYzlTVG0h9P-K-I"
              name="choice"
              onChange={[Function]}
              type="radio"
              value="choice"
            />
            <label
              className="text text_type_button label"
              htmlFor="igRc8KcYzlTVG0h9P-K-I"
            >
              Выбор
            </label>
          </div>
          <div
            className="content "
          >
            <input
              checked={false}
              className="input"
              id="GP1Qynj_X52rH7uSWQU2T"
              name="vial"
              onChange={[Function]}
              type="radio"
              value="vial"
            />
            <label
              className="text text_type_button label"
              htmlFor="GP1Qynj_X52rH7uSWQU2T"
            >
              Пузырек
            </label>
          </div>
        </div>
        <div
          className="flex button"
        >
          <button
            className="text text_type_button text_color_primary button average false "
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <svg
              fill="inherit"
              height="12"
              viewBox="0 0 20 12"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 0C0.447715 0 0 0.447715 0 1C0 1.55228 0.447715 2 1 2H11C11.5523 2 12 1.55228 12 1C12 0.447715 11.5523 0 11 0H1ZM0 6C0 5.44772 0.447715 5 1 5H15C15.5523 5 16 5.44772 16 6C16 6.55228 15.5523 7 15 7H1C0.447715 7 0 6.55228 0 6ZM0 11C0 10.4477 0.447715 10 1 10H19C19.5523 10 20 10.4477 20 11C20 11.5523 19.5523 12 19 12H1C0.447716 12 0 11.5523 0 11Z"
                fill="inherit"
                fillOpacity="0.85"
              />
            </svg>
            <p
              className="text ml-5"
            >
              По возрастанию
            </p>
          </button>
          <button
            className="text text_type_button text_color_primary button average false "
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <svg
              fill="inherit"
              height="12"
              viewBox="0 0 20 12"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M0 1C0 0.44772 0.44772 0 1 0H19C19.5523 0 20 0.44772 20 1C20 1.55228 19.5523 2 19 2H1C0.44772 2 0 1.55228 0 1ZM0 6C0 5.4477 0.44772 5 1 5H15C15.5523 5 16 5.4477 16 6C16 6.5523 15.5523 7 15 7H1C0.44772 7 0 6.5523 0 6ZM1 10C0.44772 10 0 10.4477 0 11C0 11.5523 0.44772 12 1 12H11C11.5523 12 12 11.5523 12 11C12 10.4477 11.5523 10 11 10H1Z"
                fill="inherit"
                fillOpacity="0.85"
                fillRule="evenodd"
              />
            </svg>
            <p
              className="text ml-5"
            >
              По убыванию
            </p>
          </button>
        </div>
        <div
          className="button_rigth"
        >
          <button
            className="text text_type_button text_color_primary button average false "
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <p
              className="text undefined"
            >
              Новый массив
            </p>
          </button>
        </div>
      </div>
      <div
        className="flex column"
      />
    </div>
  </div>
  <p
    className="text text_type_column text_color_input mt-14 copyright"
  >
    © Сделано в Практикуме.
  </p>
</main>
`;
